### PR TITLE
added github action

### PR DESCRIPTION
**Pull request by ArSiu**

### CODE REVIEW

**Comment by ArSiu** [comment]
@Wectro20 pls review

File: .github/workflows/test-action.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - '*'
```
Comment (ajax-gnatiuk-s) [inline .github/workflows/test-action.yaml]:
```suggestion
      - 'main'
```

File: .github/workflows/test-action.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Tests
```
Comment (ajax-gnatiuk-s) [inline .github/workflows/test-action.yaml]: let's rename to something like `Run checks`

Comment (ajax-gnatiuk-s) [inline .github/workflows/test-action.yaml]: rename the file. `test-action` is bad name

File: .github/workflows/test-action.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'corretto'
+          java-version: '17'
+
+      - name: Build and Test
+        run: ./gradlew clean test -Pprofile=test
```
Comment (ajax-gnatiuk-s) [inline .github/workflows/test-action.yaml]:
```suggestion
        run: ./gradlew build
```

File: .github/workflows/test-action.yaml (hunk 1 above)
Comment (ArSiu) [reply .github/workflows/test-action.yaml]: `+`

File: .github/workflows/test-action.yaml (hunk 2 above)
Comment (ArSiu) [reply .github/workflows/test-action.yaml]: `+` Java Build and Test Workflow

Comment (ArSiu) [reply .github/workflows/test-action.yaml]: `+`

Comment (ajax-gnatiuk-s) [reply .github/workflows/test-action.yaml]: But you haven't java in the project :))

**Review by ajax-gnatiuk-s — APPROVED** [review]

File: .github/workflows/test-action.yaml (hunk 3 above)
Comment (ArSiu) [reply .github/workflows/test-action.yaml]: `+`

File: .github/workflows/test-action.yaml (hunk 1 above)
Comment (ArSiu) [reply .github/workflows/test-action.yaml]: `+`

File: .github/workflows/test-action.yaml (hunk 2 above)
Comment (ArSiu) [reply .github/workflows/test-action.yaml]: `+`

**Pull request merged**